src/PdfSharper/Pdf.IO/Lexer.cs

```diff
@@ -176,32 +212,32 @@ public Symbol ScanNextToken()
         /// </summary>
         public byte[] ReadStream(int length)
         {
-            int pos;
+            byte[] bytes = new byte[length];
 
             // Skip illegal blanks behind �stream�.            
             MoveToNonSpace();
 
-            // Skip new line behind �stream�.
-            if (_currChar == Chars.CR)
+            int pos = _idxChar - _bufferOffset;
+            int read = 0;
+            do
             {
-                if (_nextChar == Chars.LF)
-                    pos = _idxChar + 2;
-                else
-                    pos = _idxChar + 1;
-            }
-            else
-                pos = _idxChar + 1;
+                int available = Math.Min(length - read, _bufferLength - pos);
+                Buffer.BlockCopy(_byteBuffer, pos, bytes, read, available);
+                read += available;
 
-            _pdfSteam.Position = pos;
-            byte[] bytes = new byte[length];
-            int read = _pdfSteam.Read(bytes, 0, length);
-            Debug.Assert(read == length);
-            if (bytes.Length != read)
-            {
-                Array.Resize(ref bytes, read);
-            }
-            // Synchronize idxChar etc.
-            Position = (int)_pdfSteam.Position;
+                if (read < length)
+                {
+                    _pdfStream.Position = _bufferOffset + _bufferLength;
+                    _bufferOffset = (int)_pdfStream.Position;
+                    _bufferLength = _pdfStream.Read(_byteBuffer, 0, _byteBuffer.Length);
+                    _buffer = Encoding.Default.GetChars(_byteBuffer, 0, _bufferLength);
+                    pos = 0;
+                }
+            } while (read < length);
```

Comment:
No exit condition if an invalid length was passed in (too large). It will spin infinitely determining there's nothing it can read, but it's not read its goal. Maybe throw an exception if (_bufferLength == 0) after a read operation?